### DISCUSSION TITLE
generation args

### DISCUSSION THREAD

**Discussion by shiweiba** [open-compass/opencompass#288] [View all discussions]
Hi, it's excellent work! Thanks for your contribution.

I am wondering what detailed generating arguments/strategies have been used to evaluate the open-source/APIs on the leaderboard. Is it greedy or nucleus sampling? what are the temperature and top_p?

Thanks!

**Comment by gaotongxiao** [comment]
For HuggingFace models, greedy sampling is used:
 https://github.com/InternLM/opencompass/blob/c26ecdb1b05baea7bcf34c99ea245ae68a3ada83/opencompass/models/huggingface.py#L129-L147

For other API models, we stick to their default temperature and top_p settings.
